docs/shared/concepts/customizing-inputs.md

```diff
@@ -0,0 +1,128 @@
+      "!{projectRoot}/**/*.md",
+      "!{projectRoot}/tsconfig.spec.json",
+      "!{projectRoot}/jest.config.ts",
+      "!{projectRoot}/.eslintrc.json"
+    ]
+  }
+}
+```
+
+This configuration defines three `namedInputs`.
+
+- `sharedGlobals` contains root level files that should invalidate the cache for all projects. Note that `tsconfig.base.json` and the `package.json` lock files are handled behind the scenes by Nx so that when you modify the specific properties that affect your project, the cache is invalidated.
+- `default` contains every file in the project and everything defined in `sharedGlobals`
+- `production` starts with every file in `default`, but removes `.spec.ts` files, markdown files and some project-specific config files.
+
+These names are not hard-coded. You can add your own `namedInputs` or rename these as you see fit.
+
+With these `namedInputs` in place, we can set default `inputs` for project targets in `nx.json`:
+
+```jsonc
+{
+  "targetDefaults": {
+    "build": {
+      "inputs": ["production", "^production"],
+      "dependsOn": ["^build"]
+    },
+    "test": {
+      "inputs": ["default", "^production", "{workspaceRoot}/jest.preset.js"]
+    }
+  }
+}
+```
+
+In this configuration, `build` needs to be re-run if the project's `production` files are changed or the dependencies' `production` files are changed. The `"dependsOn": ["^build"]` setting ensures that the output files of the dependencies are correct, but it does not tell Nx to re-run this project's build target when those outputs change. That's why `"^production"` is needed in the `inputs` array.
+
+Note that we are using `^production` instead of `^default` because changing test files of a dependency should not invalidate the build cache for this project.
+
+The `test` target cache is invalidated if test or production code is changed in this project (`default`) or if production code is changed in a dependency `^production`. The tests also need to be run again if the root level `jest.preset.js` file is modified.
+
+### Project Settings
+
+In the `ui-library` library, we have some `*.spec.tsx` files that we don't want to count as production files. We can modify the project configuration to account for this:
+
+```jsonc
+{
+  "namedInputs": {
+    "production": [
+      // copied this definition from nx.json
+      "default",
+      "!{projectRoot}/**/*.spec.ts",
+      "!{projectRoot}/**/*.spec.tsx", // added this line
+      "!{projectRoot}/**/*.md",
+      "!{projectRoot}/tsconfig.spec.json",
+      "!{projectRoot}/jest.config.ts",
+      "!{projectRoot}/.eslintrc.json"
+    ]
+  }
+}
+```
+
+Note that I need to copy the entire definition of `production` from `nx.json`. This entry overwrites the existing `production` entry defined in `nx.json`, but `default` and `sharedGlobals` are inherited. Now when I modify a `spec.tsx` file in `ui-library`, the build for `ui-library` is unaffected and the build for `react-app` is unaffected.
```

Comment:
@bcabanes do we want to have the inline code so bold? I find it kinda distracting. Because sometimes you'd want to highlight key passages in bold which helps parse text much faster. But with the current setting all the inline code parts are already bold and stand out.

What we have right now:
![image](https://user-images.githubusercontent.com/542458/193339338-c782f0f0-38e5-4f24-acfc-57fc7cc219e4.png)

If I want to highlight something, it doesn't really stand out:
![image](https://user-images.githubusercontent.com/542458/193339622-fa25c3ea-af2e-45bc-92ed-e08d6ae3f5e9.png)

Here's what would be more ideal:
![image](https://user-images.githubusercontent.com/542458/193339558-473ad614-7c74-49cd-b8b7-091b458d7b6f.png)

Reply:
This one is a bit more tricky since there are multiple layers of typography and component cascading. I'll look into it.

Reply:
I see. Yeah I thinking making dedicated, important parts bold in a text flow can be very powerful to increase reading speed and the ability to parse larger chunks of text. So it'd be good if we can leverage that